### DISCUSSION TITLE
[Enhancement] Plot without temp files if you have enough ram

### DISCUSSION THREAD

**Discussion by bernharl** [Chia-Network/chia-blockchain#3262] [General]
Plotting right now can wear down SSDs. If one has enough ram to support plotting, wouldn't it be nice to instead only use that for the entire plot process? I know you can use ramdisks of course, but that comes with a bit of overhead and isn't much faster than an NVMe.

**Comment by lvcivs** [comment]
Has been discussed a lot. It can be done, but the problem is you need a ton of RAM, which is expensive and only possible on server hardware (normal mainboards don't have enough RAM slots). And then you can do only 1 plot at a time, no parallel-plotting. The learn-ed opinion of the plotting masters is that it is much, much easier and cheaper to do parallel plotting on NVME SSDs.

**Comment by bernharl** [comment]
Yeah, it would obviously only be beneficial if you for some reason had access to hundreds of gigabytes of ram. But you never know.

Why no parallel-plotting?

**Comment by lvcivs** [comment]
Well, you need 332 GB RAM per plot for temp space, so if you want to do 5 parallel plots in RAM you will need  ≈ 1.6 TB RAM. Compare the price of that to the price of a single fast 2 TB NVME SSD and the RAM approach is cost prohibitive.

**Comment by bernharl** [comment]
Ah, right. I was thinking only in the technical sense.

I see your point, pretty pointless to have this feature. Maybe it would be useful in the future with cheaper RAM, but pointless to implement right now :+1:

**Comment by santimusso19** [comment]
Hi im new in this web and in this world of crypto. My doubt is, if SSD at 3500mb/s plotting takes about 4-6 hours and also have a life of determinated TB, RAM DDR3 at 1300 have 10000mb/s  x3 times than NVME and no limit to write. So for big mining rigs it wouldnt be more profitable?
